src/_pytest/_io/saferepr.py

```diff
@@ -6,8 +6,11 @@ def _format_repr_exception(exc, obj):
     exc_name = type(exc).__name__
     try:
         exc_info = str(exc)
-    except Exception:
-        exc_info = "unknown"
+    except BaseException as exc:
```

Comment:
we shouldn't catch BaseException, this'll capture `KeyboardInterrupt`, `SystemExit`, `SystemError`, etc. which *should* teardown the process

Reply:
`KeyboardInterrupt` is handled in the outer one - should be done here then also?
I've thought it would be good to abort any long-running `str()` (whatever that might be, without stopping pytest).  Will change it.

Reply:
Re "SystemError": wouldn't that be ok to handle?  That might include memory errors etc (not verified).
I'd only exclude `KeyboardInterrupt` / `SystemExit`?

Reply:
SystemError is a `Exception`.

Reply:
https://docs.python.org/3/library/exceptions.html#exception-hierarchy

Reply:
ah oops I meant `GeneratorExit` it seems :upside_down_face:

Reply:
> GeneratorExit

So you would like to crash with an INTERNALERROR then?